src/utils/details.js

```diff
@@ -27,6 +28,15 @@ function _createRequest ({
   ])
 }
 
+function validateHeaders (headers) {
+  for (const name in headers) {
+    const value = headers[name].toString()
+    if (value && value.indexOf('\n') !== -1) {
```

Comment:
I would prefer to enforce a much stricter RegEx on both `name` and `value`, to allow only alpha-numerical values, that way you also disallow messy stuff like ':' in the name, `\r`, `\0`, `\del`, etc.